compiler/luci/pass/src/RemoveUnnecessaryTransposeNetPass.cpp

```diff
@@ -260,19 +289,44 @@ bool TaggedShapeAnalyzer::can_remove_transposes(const luci::CircleTranspose *f_t
                                                 const luci::CircleReshape *m_rs,
                                                 const luci::CircleTranspose *b_tr)
 {
-  assert(loco::must_cast<luci::CircleConst *>(f_tr->perm())->dtype() == DType);
-  assert(loco::must_cast<luci::CircleConst *>(b_tr->perm())->dtype() == DType);
-
   const luci::CircleNode *in_tensor = loco::must_cast<luci::CircleNode *>(f_tr->a());
 
-  init_shape_with_tag(in_tensor);
+  // check c1
+  RET_FALSE_UNLESS(in_tensor->rank() >= b_tr->rank());
+
+  const auto front_perm = dynamic_cast<luci::CircleConst *>(f_tr->perm());
+  const auto back_perm = dynamic_cast<luci::CircleConst *>(b_tr->perm());
+
+  // check c2
+  RET_FALSE_UNLESS(front_perm != nullptr && front_perm->dtype() == DType);
+  RET_FALSE_UNLESS(back_perm != nullptr && back_perm->dtype() == DType);
+
+  const auto in_shape_v = extract_shape(in_tensor);
+  const auto front_perm_v = extract_const<DType>(front_perm);
+  const auto mid_shape_v = extract_shape(m_rs);
+  const auto back_perm_v = extract_const<DType>(back_perm);
+
+  auto all_known = [](const std::vector<int32_t> &v) -> bool {
+    for (auto i : v)
+      if (i < 0)
+        return false;
+    return true;
+  };
+
+  // check c3
+  RET_FALSE_UNLESS(all_known(in_shape_v));
+  RET_FALSE_UNLESS(all_known(front_perm_v));
+  RET_FALSE_UNLESS(all_known(mid_shape_v));
+  RET_FALSE_UNLESS(all_known(back_perm_v));
```

Comment:
This is a part that 
  * caller extracts information from each circle node
  * checks some conditions about extracted information - possible to analyze this pattern or not